rust/arrow/src/compute/kernels/cast.rs

```diff
@@ -1511,6 +1641,65 @@ mod tests {
         assert!(9.0 - c.value(4) < f64::EPSILON);
     }
 
+    #[test]
+    fn test_cast_i64_to_decimal() {
+        let a = Int64Array::from(vec![1, 2]);
+        let array = Arc::new(a) as ArrayRef;
+        let b = cast(&array, &DataType::Decimal(5, 10)).unwrap();
+        let c = b.as_any().downcast_ref::<Decimal128Array>().unwrap();
+
+        assert_eq!("1.0000000000", c.value(0).to_string());
+        assert_eq!("2.0000000000", c.value(1).to_string());
+    }
+
+    #[test]
+    fn test_cast_f64_to_decimal() {
+        let a = Float64Array::from(vec![1.0, 2.0]);
+        let array = Arc::new(a) as ArrayRef;
+        let b = cast(&array, &DataType::Decimal(5, 10)).unwrap();
```

Comment:
My understanding is that scale should be <= precision, such that we can't have `Decimal(5, 10)`. Is this correct, or is it me not understanding the rules?

Reply:
Not, it can be. I don't see any limitations.